### DISCUSSION TITLE
Import Routingpoints/Waypoints via GPX?)

### DISCUSSION THREAD

**Discussion by ThomasMeister123** [osmandapp/OsmAnd#14122] [Q&A]
I want generate a file with Routingpoints (no track!) on my PC. Then I will import this file in OSMAnd and will let OSMAnd calculate the route and navigate these points.
Maybe something like this:

```
<?xml version='1.0' encoding='UTF-8' standalone='yes' ?>
  <metadata>
    <name>Test</name>
  </metadata>
  <wpt lat="49.720871" lon="8.6936426">
    <name>Point A</name>
  </wpt>
  <wpt lat="49.7246895" lon="8.6374388">
    <name>Point B</name>
  </wpt>
  <wpt lat="49.7684441" lon="8.5872984">
    <name>Destination</name>
  </wpt>
</gpx>
```

But when I import this, there is only the option "follow track" but that not work (because it is NOT a track!)

How can use an external file for importing Waypoints/Routingpoints?

**Comment by ThomasMeister123** [comment]
OK, here a complete example:

GPX:
```

<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<gpx version="1.1" xmlns="http://www.topografix.com/GPX/1/1" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://www.topografix.com/GPX/1/1 http://www.topografix.com/GPX/1/1/gpx.xsd">
 <rte>
    <name>Etappe 1</name>
     <rtept lon="13.377119023570234" lat="52.51237129726361">
       <name>xx Berlin</name>
     </rtept>
     <rtept lon="13.031509530829743" lat="52.314536971302736">
       <name>xx Michendorf </name>
     </rtept>
     <rtept lon="13.777842804344564" lat="52.472280143640674">
       <name>xx R&#252;dersdorf </name>
     </rtept>
     <rtept lon="13.377119023570234" lat="52.51237129726361">
       <name>xx Berlin</name>
     </rtept>
 </rte>
</gpx>
```

I import this file in OSMAnd. The four points ar visible in the map (linked with a dotted red line) (1)
![01](https://user-images.githubusercontent.com/62500258/164982345-cc4215ba-6e22-4a7d-943e-c332c5629fe7.jpg)

When I use the "Navigate-Button" (2) the calculated route goes from the first Point to the last Point:

![02](https://user-images.githubusercontent.com/62500258/164982349-2aa62b8f-68e1-4401-b731-346ad232c00c.jpg)

**Reply by vshcherb** [reply]
You need to connect points with "Driving" profile: Follow Track ->  Connect Route Points. There is an issue which needs to be fixed in 4.2 https://github.com/osmandapp/OsmAnd/issues/14126

**Comment by ThomasMeister123** [comment]
Thank you. But it doesnt work.
I selected "follow track" and then this: 
<table style="width:100%">
  <tr>
<th><img src="https://user-images.githubusercontent.com/62500258/165354729-71847f47-7703-4bfa-a24e-353fde01bf06.jpg"  width="300" height="auto"/> </td>
  </tr>
</table>


Then OSMAnd is calculating, and calculating, and calculating, and calculating....

<table style="width:100%">
  <tr>
<th><img src="https://user-images.githubusercontent.com/62500258/165354803-517f93f1-503a-4bd2-920c-d6e4241da853.jpg"  width="300" height="auto"/> </td>
  </tr>
</table>







No Route, just this ("OSMAnd could not find a route, maybe maps are missing"):

<table style="width:100%">
  <tr>
<th><img src="https://user-images.githubusercontent.com/62500258/165354867-b5ba751f-6703-4bc2-a5d2-cbdc1f7e1f97.jpg"  width="300" height="auto"/> </td>
  </tr>
</table>






But maps are loaded:


<table style="width:100%">
  <tr>
<th><img src="https://user-images.githubusercontent.com/62500258/165354991-2cb24383-3138-49a1-aae5-1a42a9a34491.jpg"  width="300" height="auto"/> </td>
  </tr>
</table>

**Comment by ThomasMeister123** [comment]
OK. Sometimes (one out of ten!) after a very very very long time it works.
But this would be much better: when you select a GPX with trackpoints these points are in the "original" list of trackpoints (the same list, like when you add trackpoints manually

**Comment by ThomasMeister123** [comment]
Today I had to painfully realize that not even GPX files exported from OSMAnd work perfectly after "re-import":
I created a route with several waypoints and then exported it as a GPX to use the route on another phone. The navigation works after the import, but:
1. no distance/time to the next waypoint is displayed (on the right in the panel) (just the whole route)
2. one cannot simply remove an intermediate point
3. the navigation announcements are strange: at every junction it is said that you should continue straight ahead (more precisely: "keep left"). This is not the case with "normal" navigation.

Why are the points in the GPX not included in the list as "real" intermediate points/waypoints?

**Comment by ThomasMeister123** [comment]
So, my question: How can I save a planned trip with multiple waypoints, so that I can "load" the route and have all waypoints in the waypoint-list (e.g. I planned 5 different Routes foy holiday)
(I asked NOT for export/import; I will store different routes WITHIN one OSMAnd!)

**Comment by ThomasMeister123** [comment]
But there also the COMPLETE track is in the GPX. I want create a file that only includes 5 Points (for example). Then I want open/Import that file in OSMAnd and use this Points as Navigationwaypoints (and let OSMAND do the route-calculation)
